### ISSUE TITLE
Python File Menu

### ISSUE THREAD

**Issue by SanjeevStephan** [SanjeevStephan/pyscripts#1]
# PyFileMenu : Filter 'py'
File on github :  [pyFileMenu.py](https://github.com/SanjeevStephan/pyscripts/blob/main/pyFileMenu/pyFileMenu.py)

```
import os

# Get the list of Python files in the current directory
files = [file for file in os.listdir('.') if file.endswith('.py')]

# Display the list of files in the format [item number] filename.py
for i, file in enumerate(files):
    print("[{}] {}".format(i+1, file))

# Prompt the user to input the item number of the file they want to execute
while True:
    try:
        item_number = int(input("Enter the item number of the file you want to execute: "))
        if item_number < 1 or item_number > len(files):
            raise ValueError()
        break
    except ValueError:
        print("Invalid input, please enter a valid item number.")

# Execute the selected file
selected_file = files[item_number-1]
print("Executing file:", selected_file)
exec(open(selected_file).read())

```
## How the Script Work ?

This script is a basic Python program that allows the user to choose a Python file from the current directory and executes it.

### Here's a breakdown of the script:

1. **import os** - imports the **os** module, which provides a way to interact with the file system.
2. **files = [file for file in os.listdir('.') if file.endswith('.py')]** - uses a list comprehension to create a list of Python files in the current directory. The **os.listdir()** function returns a list of all files and directories in the current directory, and the if statement filters the list to include only files that end with the **.py** extension.
3. **for i, file in enumerate(files):** - uses a **for** loop to iterate over each file in the **files** list and assigns an index number to each file using the **enumerate()** function.
4. **print("[{}] {}".format(i+1, file))** - prints each file in the format **[item number] filename.py** to the console.
5. **while True:** - starts a **while** loop that will continue until the user enters a valid input.
6. **try**: - attempts to execute the code inside the **try** block.
7. **item_number = int(input("Enter the item number of the file you want to execute: "))** - prompts the user to enter the item number of the file they want to execute and converts the input to an integer.
8. **if item_number < 1 or item_number > len(files):** - checks if the user input is within the valid range of item numbers.
9. **raise ValueError()** - raises a **ValueError** exception if the user input is invalid.
10. **break** - breaks out of the **while** loop if the user input is valid.
11. **except ValueError**: - handles the **ValueError** exception if it is raised.
12. **print("Invalid input, please enter a valid item number.")** - prints an error message to the console if the user input is invalid.
13. **selected_file = files[item_number-1]** - selects the file that corresponds to the user's input by indexing into the files list.
14. **print("Executing file:", selected_file)** - prints the name of the selected file to the console.
15. **exec(open(selected_file).read())** - executes the selected file by opening it as a text file, reading its contents, and passing the contents to the **exec()** function, which executes the code in the file. Note that using the **exec()** function can be dangerous, as it allows arbitrary code execution, so it should be used with caution.

**Comment by SanjeevStephan** [comment]
# PyFileMenu : Exclude Substring 'random'
File on github : [pyFileMenu_exclude_only_substring_random.py](https://github.com/SanjeevStephan/pyscripts/blob/main/pyFileMenu/pyFileMenu_exclude_only_substring_random.py)
To exclude only those files with 'random' substring, you can modify the list comprehension statement to include an additional condition using the not in operator. 

Here's the modified code:
```
import os

# Get the list of Python files in the current directory that don't contain 'random' in their filenames
files = [file for file in os.listdir('.') if file.endswith('.py') and 'random' not in file and file != os.path.basename(__file__)]

# Display the list of files in the format [item number] filename.py
for i, file in enumerate(files):
    print("[{}] {}".format(i+1, file))

# Prompt the user to input the item number of the file they want to execute
while True:
    try:
        item_number = int(input("Enter the item number of the file you want to execute: "))
        if item_number < 1 or item_number > len(files):
            raise ValueError()
        break
    except ValueError:
        print("Invalid input, please enter a valid item number.")

# Execute the selected file
selected_file = files[item_number-1]
print("Executing file:", selected_file)
exec(open(selected_file).read())

```

## Changes Made

Get the list of Python files in the current directory that don't contain 'random' in their filenames

    files = [file for file in os.listdir('.') if file.endswith('.py') and 'random' not in file and file != os.path.basename(__file__)]


This line of code creates a list called files containing the names of all Python files in the current directory, with the following conditions:

1. **file.endswith('.py')** checks if the file has a .py extension.
2. '**random' not in file** checks if the substring "random" is not present in the file name. This means that any file name that contains the word "random" will be excluded from the list.
3. **file != os.path.basename(__file__)** checks if the file name is not the name of the current Python script being executed. This ensures that the current script file is not included in the list of files.

To summarize, the **files** list only includes Python files in the current directory that meet all three of these conditions: have a **.py** extension, do not contain the substring "random" in their name, and are not the same as the script file being executed.

**Comment by SanjeevStephan** [comment]
# pyFileMenu : Include Substring 'random'
File on github : [pyFileMenu_include_only_substring_random.py ](https://github.com/SanjeevStephan/pyscripts/blob/main/pyFileMenu/pyFileMenu_include_only_substring_random.py)

If you want to include only the files that contain the 'random' substring, you can modify the list comprehension by using the **in** operator instead of **not in**. Here's the modified code:

```
import os

# Get the list of Python files in the current directory that contain 'random' in their filenames
files = [file for file in os.listdir('.') if file.endswith('.py') and 'random' in file and file != os.path.basename(__file__)]

# Display the list of files in the format [item number] filename.py
for i, file in enumerate(files):
    print("[{}] {}".format(i+1, file))

# Prompt the user to input the item number of the file they want to execute
while True:
    try:
        item_number = int(input("Enter the item number of the file you want to execute: "))
        if item_number < 1 or item_number > len(files):
            raise ValueError()
        break
    except ValueError:
        print("Invalid input, please enter a valid item number.")

# Execute the selected file
selected_file = files[item_number-1]
print("Executing file:", selected_file)
exec(open(selected_file).read())

```
In this modified code, the if statement checks if the 'random' substring is present in the filename using the in operator, so only files that contain the substring will be included in the files list.

## Changes Made
Get the list of Python files in the current directory that contain 'random' in their filenames

    files = [file for file in os.listdir('.') if file.endswith('.py') and 'random' in file and file != os.path.basename(__file__)]

This line of code creates a list called files containing the names of all Python files in the current directory, with the following conditions:

1. **file.endswith('.py')** checks if the file has a .py extension.
2. **'random' in file** checks if the substring "random" is present in the file name. This means that only file names that contain the word "random" will be included in the list.
3. **file != os.path.basename(__file__)** checks if the file name is not the name of the current Python script being executed. This ensures that the current script file is not included in the list of files.

To summarize, the **files** list only includes Python files in the current directory that meet all three of these conditions: have a .py extension, contain the substring "random" in their name, and are not the same as the script file being executed.
